src/NuGet.Core/NuGet.Protocol/Strings.resx

```diff
@@ -530,4 +530,9 @@ The "s" should be localized to the abbreviation for seconds.</comment>
 {1} - number of vulerabilitiy files reported
 {2} - max count allowed</comment>
   </data>
-</root>
+  <data name="Error_HttpServiceIndexUsage" xml:space="preserve">
+    <value>You have used a non HTTPS service index: '{0}'. NuGet requires HTTPS sources. To use HTTP sources, you must explicitly set 'allowInsecureConnections' to true in your NuGet.Config file. Please refer to https://aka.ms/nuget-https-everywhere for more information.</value>
```

Comment:
This message is saying that the service index is not https, but where this message is being used is checking the service index's entries, not the service index itself.

```suggestion
    <value>There are non-HTTPS resources defined by your package source: '{0}'. NuGet requires HTTPS endpoints. To allow unencrypted HTTP, you must explicitly set 'allowInsecureConnections' to true in your NuGet.Config file, or talk to the administrator of your source and ask them to migrate to a secure HTTPS URL. Please refer to https://aka.ms/nuget-https-everywhere for more information.</value>
```